example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,204 @@
+        {
+            int halfFilterSize = filterSize / 2;
+            TElem result = 0.0f;
+            // Calculate sum of multiplications of corresponding elements
+            for(size_t i = 0; i < filterSize; ++i)
+            {
+                int inputIndex = linearizedGlobalThreadIdx - halfFilterSize + i;
+                if(inputIndex >= 0 && inputIndex < inputSize)
+                {
+                    result += input[inputIndex] * filter[i];
+                }
+            }
+            output[linearizedGlobalThreadIdx] = result;
+        }
+    }
+};
+
+/**
+ * @brief Function template to compare floating point values
+ *
+ */
+template<typename FloatingType>
+typename std::enable_if<std::is_floating_point<FloatingType>::value, bool>::type FuzzyEqual(
+    FloatingType a,
+    FloatingType b)
+{
+    return std::fabs(a - b) < std::numeric_limits<FloatingType>::epsilon() * 10;
+}
+
+auto main() -> int
+{
+// Fallback for the CI with disabled sequential backend
+#if defined(ALPAKA_CI) && !defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+    return EXIT_SUCCESS;
+#else
+    // Define the index domain
+    using Dim = alpaka::DimInt<1u>;
+    // Index type
+    using Idx = std::size_t;
+
+    // Define the accelerator
+    using DevAcc = alpaka::ExampleDefaultAcc<Dim, Idx>;
+    using QueueProperty = alpaka::Blocking;
+    using QueueAcc = alpaka::Queue<DevAcc, QueueProperty>;
+    using DataType = float;
+
+    std::cout << "Using alpaka accelerator: " << alpaka::getAccName<DevAcc>() << std::endl;
+
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+
+    // Select a device
+    auto const platformAcc = alpaka::Platform<DevAcc>{};
+    auto const devAcc = alpaka::getDevByIdx(platformAcc, 0);
+
+    // Create a queue on the device
+    QueueAcc queue(devAcc);
+
+    // Allocate memory host input
+    auto hostInputMemory = alpaka::allocBuf<DataType, Idx>(devHost, InputSize);
+    DataType* nativeHostInputMemory = alpaka::getPtrNative(hostInputMemory);
+
+    // Fill array with data
+    for(size_t i = 0; i < InputSize; i++)
+        nativeHostInputMemory[i] = static_cast<DataType>(i + 1);
+
+    // Allocate memory host filter
+    auto hostFilterMemory = alpaka::allocBuf<DataType, Idx>(devHost, FilterSize);
+    DataType* nativeHostFilterMemory = alpaka::getPtrNative(hostFilterMemory);
+
+    // Fill array with any data
+    for(size_t i = 0; i < FilterSize; i++)
+        nativeHostFilterMemory[i] = static_cast<DataType>(i + 1) / 10.0f;
+
+    // Allocate memory in device
+    alpaka::Buf<DevAcc, DataType, Dim, Idx> inputDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, InputSize);
+    alpaka::Buf<DevAcc, DataType, Dim, Idx> filterDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, FilterSize);
+    alpaka::Buf<DevAcc, DataType, Dim, Idx> outputDeviceMemory
+        = alpaka::allocBuf<DataType, Idx>(devAcc, static_cast<Idx>(InputSize));
```

Comment:
I don't think the buffer type helps here. I would have the compiler deduce it:
```suggestion
    auto inputDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, InputSize);
    auto filterDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, FilterSize);
    auto outputDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, static_cast<Idx>(InputSize));
```

Reply:
I used an alias like in almost all examples. I remember Rene was talking about failures of some of the compilers.